integration_tests/src/main/python/string_test.py

```diff
@@ -779,4 +779,13 @@ def test_like_complex_escape():
                 'a like "\\%SystemDrive\\%\\\\\\\\Users%"',
                 'a like "_oo"'),
             conf={'spark.sql.parser.escapedStringLiterals': 'true'})
- 
+
+
+@pytest.mark.parametrize('from_base', [10], ids=idfn)
+@pytest.mark.parametrize('to_base', [16], ids=idfn)
+@pytest.mark.parametrize('pattern', ['[0-9]{1,18}', '[0-9a-fA-F]{1,15}'])
```

Comment:
What about error cases? Should we have a few patterns that do not match?

Reply:
I tested with mk_str_gen and special cases as well. This is still a WIP, will make sure to add it back